src/roadmap/models.py

```diff
@@ -48,6 +51,39 @@ class System(BaseModel):
     lifecycle_type: LifecycleType
 
 
+    @model_validator(mode='after')
+    def update_support_status(self):
+        """Validator for setting support status.
+        Expected types/values of start_date and end_date are:
+            - str(Unknown)
+            - None
+            - date(YYYY-MM-DD)
+        """
+        today = date.today()
+
+        if self.release_date != "Unknown" and self.release_date is not None:
+            if self.release_date > today:
+                self.support_status = SupportStatus.upcoming
+                return self
+
+        if self.retirement_date != "Unknown" and self.retirement_date is not None:
+            if self.retirement_date < today:
+                self.support_status = SupportStatus.retired
+                return self
+            
+            six_months_date = self.retirement_date - timedelta(days=180)
+            if six_months_date <= today:
+                self.support_status = SupportStatus.six_months
+            else:
+                self.support_status = SupportStatus.supported
+        
+        else:
+            self.support_status = SupportStatus.unknown
+            return self
+
+        return self
```

Comment:
can this ever be hit? It looks like the "else" above would catch it

Reply:
Yeah, that's right. Fixed, pushing also with the unit tests. Thanks!

Reply:
It's needed: https://github.com/RedHatInsights/digital-roadmap-backend/pull/69#discussion_r2017541080 :D

(I removed it different file, in the models I forget to do it, but the code is mostly the same)